### PR TITLE
Fix team members API

### LINKED ISSUE #6704: Members of a team in org can view detail of other teams (in the same org) via api
- Gitea version (or commit ref): 1.8.0
- Git version: 2.20.1
- Operating system: linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes ( https://rucciva-peasant:mC2yjUj5K5gcFPP@try.gitea.io/api/v1/teams/2633/members)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

If a user is added to a team in organization, the user could obtains information on other teams in the same organization via api (`GET /teams/{id}` and `GET /teams/{id}/*`). Meanwhile, accessing other team's page via web ui is forbidden

### PR DESCRIPTION
**Pull request by lunny**
will fix #6704

### CODE REVIEW

**Comment by zeripath** [comment]
Integration tests need to be updated

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -286,6 +286,40 @@ func reqOrgOwnership() macaron.Handler {
...
+		if ctx.Context.IsUserSiteAdmin() {
+			return
+		}
+		if ctx.Org.Team == nil {
+			ctx.Error(500, "", "reqTeamMembership: unprepared context")
+			return
+		}
+
+		var orgID = ctx.Org.Team.OrgID
+		isOwner, err := models.IsOrganizationOwner(orgID, ctx.User.ID)
```
Comment (lafriks) [inline routers/api/v1/api.go:301]: Why do you need to check for owner? Owner is also a team member

Comment (lunny) [reply routers/api/v1/api.go:301]: Member of Owner Team should visit all other team's pages, So we should know if the User is on the owner team of this organization.

**Review by techknowlogick — APPROVED** [review]

File: routers/api/v1/api.go [hunk 2]
```diff
@@ -286,6 +286,43 @@ func reqOrgOwnership() macaron.Handler {
...
+
+		if isTeamMember, err := models.IsTeamMember(orgID, ctx.Org.Team.ID, ctx.User.ID); err != nil {
+			ctx.Error(500, "IsTeamMember", err)
+			return
+		} else if !isTeamMember {
+			isOrgMember, err := models.IsOrganizationMember(orgID, ctx.User.ID)
+			if err != nil {
+				ctx.Error(500, "IsOrganizationMember", err)
+			} else if isOrgMember {
+				ctx.Error(403, "", "Must be the team member")
```
Comment (mrsdizzie) [inline routers/api/v1/api.go]:
```suggestion
				ctx.Error(403, "", "Must be a team member")
```

Comment (mrsdizzie) [reply routers/api/v1/api.go]: otherwise LGTM and works as expected in testing thanks!

**Review by mrsdizzie — APPROVED** [review]

**Comment by techknowlogick** [comment]
@lunny please backport 😄

**Comment by lunny** [comment]
@techknowlogick done.

**Pull request merged**